pkgs/development/libraries/quarto/default.nix

```diff
@@ -0,0 +1,73 @@
+{ stdenv
+, lib
+, pandoc
+, esbuild
+, deno
+, fetchurl
+, nodePackages
+, rWrapper
+, rPackages
+, makeWrapper
+, python3
+, python3Packages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "quarto";
+  version = "1.1.189";
+  src = fetchurl {
+    url = "https://github.com/quarto-dev/quarto-cli/releases/download/v${version}/quarto-${version}-linux-amd64.tar.gz";
+    sha256 = "1a3xsgqdccm4ky1xjnin1idpp8gsansskq37c00mrxz1raxn1mi7";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  patches = [
+      ./fix-deno-path.patch
+  ];
+
+  preFixup = ''
+      wrapProgram $out/bin/quarto \
+        --prefix PATH : ${lib.makeBinPath [ deno ]} \
+        --prefix QUARTO_PANDOC : ${lib.makeBinPath [ pandoc ]}/pandoc \
+        --prefix QUARTO_ESBUILD : ${lib.makeBinPath [ esbuild ]}/esbuild \
+        --prefix QUARTO_DART_SASS : ${lib.makeBinPath [ nodePackages.sass ]}/sass \
+        --prefix QUARTO_R : ${lib.makeBinPath [ (rWrapper.override { packages = [ rPackages.rmarkdown ]; }) ]}/R \
+        --prefix QUARTO_PYTHON : ${lib.makeBinPath [ (python3.withPackages (ps: with ps; with python3Packages; [
+                                                                                                                 jupyter
+                                                                                                                 ipython
+                                                                                                               ])) ]}/python3
+  '';
+
+  installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin
+      mkdir -p $out/share
+
+      rm -r bin/tools
+      mv bin/vendor/import_map.json bin
```

Comment:
What does import_map.json have to do in $out/bin?

Reply:
Yes that is wrong, it does not relate to the original structure.

Reply:
Fixed in https://github.com/NixOS/nixpkgs/pull/186697/commits/556de34b04f9719122e2053d2ecb27c63086ca4d